flycheck.el

```diff
@@ -2159,10 +2162,43 @@ into the verification results."
         (princ (make-string (- message-column (current-column)) ?\ ))
         (let ((message (flycheck-verification-result-message result))
               (face (flycheck-verification-result-face result)))
-          (insert (propertize message 'face face)))
+          ;; If face is nil, using propertize erases the face already contained
+          ;; by the message.  We don't want that, since this would remove the
+          ;; button face from the checker chain result.
+          (if face
+              (insert (propertize message 'face face))
+            (insert message)))
```

Comment:
But when face is non-nil, won't it override the existing face? Maybe you want `font-lock-append-text-property`?

Reply:
Oh I think I understand now.  Just a small change then:
`(insert (if face (propertize message …) message))` (no need to repeat the insert)